modules/renter/contractor/contractor.go

```diff
@@ -140,14 +150,9 @@ func (c *Contractor) CurrentPeriod() types.BlockHeight {
 // ResolveID returns the ID of the most recent renewal of id.
 func (c *Contractor) ResolveID(id types.FileContractID) types.FileContractID {
 	c.mu.RLock()
-	defer c.mu.RUnlock()
-
-	newID, exists := c.renewedIDs[id]
-	for exists {
-		id = newID
-		newID, exists = c.renewedIDs[id]
-	}
-	return id
+	newID := c.resolveID(id)
+	c.mu.RUnlock()
```

Comment:
Hmm, I think I prefer `defer RUnlock`, even though this is now a trivial method.

Reply:
Using defer has nontrivial performance penalties, amazingly enough. It's tricky because `defer` will run even if there's a panic somewhere in the call stack, and that adds a bunch of overhead.